### ISSUE TITLE
Implement K-Fold cross-validation to train the U-Net

### ISSUE THREAD

**Issue by MelanieLu** [neuropoly/deep_active_learning#6]
At the moment, the training set, validation set and test set are fixed, and new annotated patches are added only to the training set after each active-learning iteration, leaving the validation set unchanged. 

Using K-fold cross validation would allow to make use of all the samples to train the network, especially when few data samples are available. 

One potential issue coming with the K-fold cross validation is the extension of the training time.
